tests/baselines/reference/typeFromPrototypeAssignment4.js

```diff
@@ -0,0 +1,39 @@
+//// [a.js]
+function Multimap4() {
+  this._map = {};
+};
+
+Multimap4["prototype"] = {
+  /**
+   * @param {string} key
+   * @returns {number} the value ok
+   */
+  get(key) {
+    return this._map[key + ''];
+  }
+};
+
+Multimap4["prototype"]["add-on"] = function() {};
+Multimap4["prototype"]["addon"] = function() {};
+
+const map4 = new Multimap4();
+map4.get("");
+map4["add-on"]();
+map4.addon();
+
+
+
+
+//// [a.d.ts]
+declare function Multimap4(): void;
+declare class Multimap4 {
+    _map: {};
+    "add-on"(): void;

```

Comment:
❤️ 

A element-access-based field with leading `__` would be nice, just to ensure everything's escaped correctly.

Reply:
Done 👍